apps/api_web/lib/api_web/plugs/rate_limiter_concurrent.ex

```diff
@@ -0,0 +1,54 @@
+defmodule ApiWeb.Plugs.RateLimiterConcurrent do
+  @moduledoc """
+  Plug to invoke the concurrent rate limiter.
+  """
+
+  import Plug.Conn
+  import Phoenix.Controller, only: [render: 3, put_view: 2]
+
+  alias ApiWeb.RateLimiter.RateLimiterConcurrent
+
+  @rate_limit_concurrent_config Application.compile_env!(:api_web, :rate_limiter_concurrent)
+
+  def init(opts), do: opts
+
+  def call(conn, _opts) do
+    if enabled?() do
```

Comment:
suggestion: can we change this to check the rate limit and log the result if `memcache` is true, but only enforce the rate limit if `enabled` is true? I'm thinking about how we roll this out, and I don't think there's a good way to understand the expected values for some of our higher-throughput applications (dotcom, screens) without getting some benchmark data first.

Reply:
I have added two additional feature flags: `:log_statistics` and `:limit_users` to support these behaviors, and implemented a log statement w/ API key + current state, and added :limit_users == true to the condition that actually blocks the user from accessing until their cooldown.